tests/e2e/test_rosacli_cluster.go

```diff
@@ -1642,6 +1643,144 @@ var _ = Describe("HCP cluster creation negative testing",
 						ContainSubstring(
 							"ERR: Additional Allowed Principals is supported only for Hosted Control Planes"))
 			})
+
+		It("HCP cluster creation subnets validation - [id:72538]",
+			labels.High, labels.Runtime.Day1Negative,
+			func() {
+				if profile.ClusterConfig.Private {
+					rosalCommand.DeleteFlag("--private", false)
+					rosalCommand.DeleteFlag("--private-link", false)
+				}
+				if profile.ClusterConfig.Autoscale {
+					Skip("Test do not work with autoscaling enabled")
+				}
+				log.Logger.Info(profile.Name)
+				log.Logger.Info(strings.Split(rosalCommand.GetFullCommand(), " "))
+
+				clusterName := "ocp-72538"
+				vpcName := "vpc-72538"
+				region := "us-west-2"
+				availabilityZone := "us-west-2a"
+				replacingFlags := map[string]string{
+					"-c":              clusterName,
+					"--cluster-name":  clusterName,
+					"--domain-prefix": clusterName,
+					"--region":        region,
+				}
+				rosalCommand.ReplaceFlagValue(replacingFlags)
+				if rosalCommand.CheckFlagExist("--subnet-ids") {
+					rosalCommand.DeleteFlag("--subnet-ids", true)
+				}
+
+				By("Prepare a vpc for the testing")
+				vpc, err := vpc_client.PrepareVPC(vpcName, region, "", false, "")
+				Expect(err).ToNot(HaveOccurred())
+				defer vpc.DeleteVPCChain()
+
+				subnetMap, err := profilehandler.PrepareSubnets(vpc, region, []string{}, true)
+				Expect(err).ToNot(HaveOccurred())
+
+				additionalPrivateSubnet, err := vpc.CreatePrivateSubnet(availabilityZone, true)
```

Comment:
I would initialize `availabilityZone` with the AZ from  the first subnet in `subnetMap`. That way you make sure you have a duplicated AZ